pjsip/src/pjsip/sip_transport.c

```diff
@@ -1466,11 +1466,14 @@ PJ_DEF(pj_status_t) pjsip_transport_shutdown2(pjsip_transport *tp,
     pj_lock_acquire(tp->lock);
 
     mgr = tp->tpmgr;
-    pj_lock_acquire(mgr->lock);
+
+    // Acquiring manager lock after transport lock may cause deadlock.
+    // And it does not seem necessary as well.
+    //pj_lock_acquire(mgr->lock);
```

Comment:
In L1110, there is a comment that we hold the manager's lock to avoid race with `pjsip_tpmgr_acquire_transport2()`.

So is it safe for this scenario?

Reply:
Yes, I saw that too. Unfortunately it does not explain how the race condition may cause issues, e.g: when `pjsip_tpmgr_acquire_transport2()` is executed first and returns a transport, but right after that by other thread, the transport flag `is_shutdown/destroyed` is set, eventually the app will still get a transport that is being shutdown?